engine/src/main/java/org/entando/entando/aps/system/services/user/UserService.java

```diff
@@ -180,28 +185,53 @@ public PagedMetadata<UserDto> getUsers(RestListRequest requestList, String withP
             //transforms the filters by overriding the key specified in the request with the correct one known by the dto
             List<FieldSearchFilter> filters = new ArrayList<>(requestList.buildFieldSearchFilters());
             List<EntitySearchFilter> entityFilters = new ArrayList<>(requestList.buildEntitySearchFilters());
+            boolean filterOnProfileAttributes = !entityFilters.isEmpty();
             filters.stream()
                     .filter(i -> ((i.getKey() != null) && (UserDto.getEntityFieldName(i.getKey()) != null)))
                     .forEach(i -> i.setKey(UserDto.getEntityFieldName(i.getKey())));
-            List<String> userNames = null;
+            List<String> userNames;
             List<UserDetails> users = new ArrayList<>(),
                     allUsers = new ArrayList<>();
 
             //username filter
-            List<FieldSearchFilter> usernameFilter = filters.stream().filter(filter
-                    -> filter.getValue() != null && filter.getKey().equals("username")).collect(Collectors.toList());
-            if (usernameFilter.size() > 0) {
-                String text = (String) filters.get(0).getValue();
+            Optional<String> usernameFilter = filters.stream().filter(filter
+                    -> filter.getValue() != null && filter.getKey().equals("username"))
+                    .map(filter -> (String) filter.getValue()).findFirst();
+
+            if (usernameFilter.isPresent()) {
+                String text = usernameFilter.get();
                 userNames = this.getUserManager().searchUsernames(text);
+                EntitySearchFilter<String> profileUsernameFilter = new EntitySearchFilter<>(
+                        "username", false, text, true, LikeOptionType.COMPLETE);
+                entityFilters.add(profileUsernameFilter);
             } else {
                 userNames = this.getUserManager().getUsernames();
             }
-            userNames.forEach(username -> allUsers.add(this.loadUser(username)));
 
-            // Profile and attributes filters
-            users.addAll(allUsers.stream().filter(user
-                    -> (withProfile == null || withProfile.equals("1"))
-                    && checkAttributesFilter(filters, entityFilters, user)).collect(Collectors.toList()));
+            List<String> profiles = this.getUserProfileManager().searchId(entityFilters.toArray(EntitySearchFilter[]::new));
```

Comment:
maybe in general some log at debug could be usefull

Reply:
done